From b6d00556b6f35fa76bc8ef7d34d05a4bebcc9e55 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sun, 22 Nov 2020 19:24:46 +0100
Subject: [PATCH 1/1] fs: fat: dentry iterator for fill_dir_slot()

For reusing deleted directory entries we have to adjust the function called
to step to the next directory entry.

This patch alone is not enough to actually reuse deleted directory entries
as the fill_dir_slot() is still called with first never used directory
entry.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 fs/fat/fat_write.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/fat/fat_write.c b/fs/fat/fat_write.c
index c4deb6fed2..c7ee514a1e 100644
--- a/fs/fat/fat_write.c
+++ b/fs/fat/fat_write.c
@@ -358,7 +358,7 @@ fill_dir_slot(fat_itr *itr, const char *l_name, const char *shortname)
 			flush_dir(itr);
 
 		/* allocate a cluster for more entries */
-		if (!fat_itr_next(itr) && !itr->dent)
+		if (!next_dent(itr) && !itr->dent)
 			if ((itr->is_root && itr->fsdata->fatsize != 32) ||
 			    new_dir_table(itr))
 				return -EIO;
-- 
2.29.2

